src/middleware/wrapSubmissionErrors.js

```diff
@@ -3,14 +3,15 @@ import { SubmissionError } from 'redux-form'
 import isPromise from 'is-promise'
 import { getOr } from 'lodash/fp'
 
-// Wrap submission results in a redux-form SubmissionError
+// Wrap submission results in a redux-form SubmissionError. 
+// Also ensures that the return value of onSubmit is a promise.
 const wrapSubmissionErrors = withPropsOnChange(
   ['onSubmit'],
   ({ onSubmit }) => {
     return {
       onSubmit: (...args) => {
         const result = onSubmit(...args)
-        if (!isPromise(result)) return result
+        if (!isPromise(result)) return Promise.resolve(result)
```

Comment:
Thoughts about renaming `wrapSubmissionErrors` or splitting this out into  a separate function? I get not wanting to have this as a separate middleware since it would then be wrapped for submission errors unnecessarily, but it is drifting towards multi-responsibility

Reply:
Yeah it's definitely close to multi responsibility. I think it's at a spot I'm comfortable with for now though. What name were you thinking?

Reply:
`wrapSubmission` or `wrapSubmissionPromise`?

Reply:
^ the naming is the trickiest bit 😛